src/Transporters/HttpTransporter.php

```diff
@@ -46,15 +54,18 @@ public function requestObject(Payload $payload): Response
     {
         $request = $payload->toRequest($this->baseUri, $this->headers, $this->queryParams);
 
-        $response = $this->sendRequest(fn (): \Psr\Http\Message\ResponseInterface => $this->client->sendRequest($request));
+        $response = $this->sendRequest(
+            fn (): \Psr\Http\Message\ResponseInterface => $this->client->sendRequest($request),
+            $request
+        );
 
         $contents = $response->getBody()->getContents();
 
         if (str_contains($response->getHeaderLine('Content-Type'), ContentType::TEXT_PLAIN->value)) {
             return Response::from($contents, $response->getHeaders());
         }
```

Comment:
The work in this PR looks good but I'm seeing intermittent requests failing due to OpenAI API flakeyness (#478). This block needs some of the precautions of `throwIfJsonError()` otherwise the `string` response data starts causing `TypeError`.

```
Status Code: 404
Content-Type: "text\/plain; charset=utf-8"
Body: "404 page not found\n"
```